streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -193,13 +193,17 @@ private void closeAndRevive(final Map<Task, Collection<TopicPartition>> taskWith
 
             try {
                 task.suspend();
+                // we need to enforce a checkpoint that removes the corrupted partitions
+                task.postCommit(true);
```

Comment:
I'm not sure this is 100% ensuring the snapshot gets done, as we have branching logic in postCommit. Do you think we should just get a helper to cleanup instead?

Reply:
Here since the task is guaranteed to be in SUSPENDED `postCommit()` is ensured to trigger checkpointing.